File: prime_slice_memory.go
package main

func main() {
	const N = 20000000

	var primes [N]int
	var head int

	for num := 2; num < N; num++ {
		isPrime := true
		for _, prime := range primes[:head] {
			if prime*prime > num {
				break
			}
			if num%prime == 0 {
				isPrime = false
				break
			}
		}

		if isPrime {
			primes[head] = num
			head++
		}
	}

	println(head)
}

//Compiled: go build -ldflags="-s -w"
//Run Time: 6.0 sec
//Memory before: 7,000 K
//Memory After: 16,900 K

File: prime_slice_memory.rs
fn main() {

  let mut array_primes: Vec<u64> = vec![0;20000000];
  let mut slice_len = 0;

  for i in 2..20000000 {
    if array_primes.iter().take_while(|&p| p * p <= i && *p != 0).all(|&p| i % p != 0) {
      array_primes[slice_len] = i; //Set value
      slice_len += 1;
    }
  }
  println!("{}", slice_len);

}

//Compiled: cargo build --release
//5.1 sec
//Memory Before: 576 K
//Memory After: 10,516 K
